logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/CallRepository.kt

```diff
@@ -3,68 +3,71 @@ package com.wire.kalium.logic.data.call
 import com.wire.kalium.logic.data.message.Message
 import com.wire.kalium.logic.data.message.MessageContent
 import com.wire.kalium.logic.data.message.PersistMessageUseCase
 import com.wire.kalium.logic.data.team.TeamRepository
 import com.wire.kalium.logic.data.user.UserId
 import com.wire.kalium.logic.data.user.UserRepository
 import com.wire.kalium.logic.data.user.toUserId
+import com.wire.kalium.logic.di.MapperProvider
 import com.wire.kalium.logic.feature.call.Call
 import com.wire.kalium.logic.feature.call.CallStatus
 import com.wire.kalium.logic.functional.Either
 import com.wire.kalium.logic.util.TimeParser
 import com.wire.kalium.logic.wrapApiRequest
+import com.wire.kalium.logic.wrapStorageRequest
 import com.wire.kalium.network.api.call.CallApi
+import com.wire.kalium.persistence.dao.ConversationEntity
+import com.wire.kalium.persistence.dao.call.CallDAO
+import com.wire.kalium.persistence.dao.call.CallEntity
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.first
-import kotlinx.coroutines.flow.map
 import kotlin.math.max
 
+@Suppress("TooManyFunctions")
 interface CallRepository {
     suspend fun getCallConfigResponse(limit: Int?): Either<CoreFailure, String>
     suspend fun connectToSFT(url: String, data: String): Either<CoreFailure, ByteArray>
-    fun updateCallProfileFlow(callProfile: CallProfile)
-    fun callsFlow(): Flow<List<Call>>
-    fun incomingCallsFlow(): Flow<List<Call>>
-    fun ongoingCallsFlow(): Flow<List<Call>>
-    fun establishedCallsFlow(): Flow<List<Call>>
-    suspend fun createCall(
-        conversationId: ConversationId,
-        status: CallStatus,
-        callerId: String,
-        isMuted: Boolean,
-        isCameraOn: Boolean
-    )
-
+    fun updateCallMetadataProfileFlow(callMetadataProfile: CallMetadataProfile)
+    fun getCallMetadataProfile(): CallMetadataProfile
+    suspend fun callsFlow(): Flow<List<Call>>
+    suspend fun incomingCallsFlow(): Flow<List<Call>>
+    suspend fun ongoingCallsFlow(): Flow<List<Call>>
+    suspend fun establishedCallsFlow(): Flow<List<Call>>
+    suspend fun createCall(conversationId: ConversationId, status: CallStatus, callerId: String, isMuted: Boolean, isCameraOn: Boolean)
     suspend fun updateCallStatusById(conversationId: String, status: CallStatus)
     fun updateIsMutedById(conversationId: String, isMuted: Boolean)
     fun updateIsCameraOnById(conversationId: String, isCameraOn: Boolean)
     fun updateCallParticipants(conversationId: String, participants: List<Participant>)
     fun updateParticipantsActiveSpeaker(conversationId: String, activeSpeakers: CallActiveSpeakers)
+    suspend fun getLastClosedCallCreatedByConversationId(conversationId: ConversationId): Flow<String?>
+
+    /**
+     * To be used only in Debug mode
+     */
+    suspend fun deleteAllCalls()
```

Comment:
Can we get more details about this ? why do we need to delete all calls for debug only ?